### DISCUSSION TITLE
CI/CD along with Bash-Script

### DISCUSSION THREAD

**Discussion by rajavishah** [bilayer-containers/bilayers#15] [Q&A] [answered]
As discussed with @gnodar01, we are developing CI/CD pipeline which takes `algorithm-names` and `interface-names` from a bash script. Eventually, the script would run for `m*n` times. (m = number of algorithms, n = number of interfaces). 

Currently, I have developed it as - Github Actions would have one `step` where it runs the whole bash script : https://github.com/bilayer-containers/bilayers/blob/2fbe84bd354fb138a2127f279b838e3a9a5822d4/.github/workflows/docker-build.yml#L32

But we can also think as bash script would only contain the names of the algorithm and interfaces whilst every `nox session` will be run in `CI/CD workflow`

Any suggestions @gnodar01 , @bethac07 ?

**Comment by gnodar01** [comment, marked as answer]
> But we can also think as bash script would only contain the names of the algorithm and interfaces whilst every nox session will be run in CI/CD workflow

Yes, I think this is preferred. Clean environment for each build.

**Comment by bethac07** [comment]
For checking which things already exist, we might crib off of the work of/talk to the [pycytominer team](https://github.com/cytomining/pycytominer/pull/377/files)
